src/parser/mod.rs

```diff
@@ -8357,10 +8358,241 @@ impl<'a> Parser<'a> {
                 }
             }
 
+            if self.dialect.supports_match_recognize()
```

Comment:
should this use the `dialect_of` check instead or is the trait method required for this feature?

Reply:
I prefer to use the trait method here because it can be reused in the test with `all_dialects_where(|d| d.supports_match_recognize()`. This way if it's added for another dialect in the future (e.g. [Trino](https://trino.io/docs/current/sql/match-recognize.html)), those tests will run without having to remember to add the new dialect to them.

Reply:
Ah I see, not sure if we have a guideline for this but feels to me like the dialect trait could get bloated quickly otherwise. Not sure though, deferring to @alamb if you have thoughts (and the PR looks good to me for a review too!)

Reply:
Yeah that's a good point. We could do something like this:

```
struct FeatureSupport {
    match_recognize: bool,
    .. options
}

trait Dialect {
    fn features() -> &'static FeatureSupport;

    // methods requiring custom code
}
```

If `DialectFeatureSupport` gets too big, we can make it hierarchical.

Reply:
I don't have a great story for this question -- it seems to me that the crate is currently inconsistent where sometimes it uses a trait method and sometimes uses the hard coded `IsDialect` .. 

I think the `FeatureSupport` idea is neat, and if we didn't already have 2 patterns it would probably be the way to go. However, given we already have 2 methods, I think adding a third would be less than ideal 

Ideally we would have a single way and I think the best thing for users of the crate is methods on Dialect (mostly so they can define their own custom dialects, which I have seen some of them do). However, I fully acknowledge that the trait will get very large for sure.

However, a wholesale refactoring of the crate is probably not feasible at this time. 

If we can get some consensus between the three of us maybe I can add something to the documentation on the preferred method (dialectof or trait method)

Reply:
The FeatureSupport proposal seems like it balances the tradeoffs closely but yeah, agree that its likely not ideal to introduce another mechanism for this. At least the Dialect trait methods route sounds very reasonable to me as a preferred method in that case since as mentioned it helps users with custom dialect setup and it makes it easier to add test coverage for dialects.